.github/workflows/release-apps.yml

```diff
@@ -0,0 +1,74 @@
+name: Release Apps
+
+on:
+  release:
+    types: [ published ]
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    name: Publish
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        app: [ legacy-api, ocean-api ]
+    steps:
+      - uses: actions/checkout@230611dbd0eb52da1e1f4f7bc8bb0c3a339fc8b7
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@27d0a4f181a40b142cce983c5393082c365d1480
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@94ab11c41e45d028884a99163086648e898eed25
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@6af3c118c8376c675363897acf1757f7a9be6583
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Resolve Tags
+        uses: actions/github-script@9ac08808f993958e9de277fe43a64532a609130e
+        id: tags
+        env:
+          APP: ${{ matrix.app }}
+        with:
+          script: return require('./.github/scripts/release-tags.js')({ context })
+          result-encoding: string
+      
+      - name: Build & Publish
+        uses: docker/build-push-action@7f9d37fa544684fb73bfe4835ed7214c255ce02b
+        with:
+          push: true
+          build-args: APP=${{ matrix.app }}
+          platforms: |
+            linux/amd64
+            linux/arm64
+          tags: ${{ steps.tags.outputs.result }}
+
+  report:
+    name: Report
+    runs-on: ubuntu-latest
+    needs: build
+    if: github.event_name == 'pull_request'
+    steps:
+      - uses: actions/checkout@230611dbd0eb52da1e1f4f7bc8bb0c3a339fc8b7
```

Comment:
Minor suggestion for optimization, since you are checking out the entire workflow with this step. Would it make sense to just have GitHub scripts with the embedded script since it's pretty short - to also keep it close and contextual together?

Just a suggestion, I am on the fence here. Either way works imo.

Reply:
While I would definitely advocate keeping things as close as possible, I think it this case it bloats the file too much. If the scripts themselves were smaller i'd probably leave them in though. In this case though I have emphasized writing it up so that any future additions to the yml stays isolated to the yml while the scripts are more abstract.

I haven't seen any performance concerns with the checkout yet, but if we see it as an issue we could move the scripts inline.

I will defer to your judgment here, but personally I think its at least more readable this way.